### PR TITLE
Improve 'getting started' for astropy.coordinates

### PR DESCRIPTION
**Pull request by taldcroft**
I've encountered several users who didn't realize ``SkyCoord`` and ``astropy.coordinates`` in general can deal with arrays. Looking at the getting started section of ``astropy.coordinates`` I realized it is a bit intimidating and goes into a fair bit of detail:

http://docs.astropy.org/en/stable/coordinates/index.html

I think things like the representation details don't belong in the getting started and instead we should emphasize things like support for arrays more (this is mentioned but very discrete at the moment)

(More generally, now that 1.0 is out, and since we now have many features, I think we need to put some effort into really showcasing this functionality more efficiently)

### CODE REVIEW

**Comment by taldcroft** [comment]
@astrofrog @eteq - docs update attached.  I pushed the array example up to the top.  I think that just by added subsection labels that it feels much less intimidating.

**Comment by taldcroft** [comment]
Note that in a few places I did actually remove content that seemed like "too much detail" in order to reduce the footprint of Getting Started.

File: docs/coordinates/index.rst [hunk 1]
```diff
@@ -35,9 +35,9 @@ equivalent::
     >>> c = SkyCoord(ra=10.625*u.degree, dec=41.2*u.degree, frame='icrs')
     >>> c = SkyCoord(10.625, 41.2, frame='icrs', unit='deg')
     >>> c = SkyCoord('00h42m30s', '+41d12m00s', frame='icrs')
-    >>> c = SkyCoord('00h42.5m', '+41d12m', frame='icrs')
-    >>> c = SkyCoord('00 42 30 +41 12 00', frame='icrs', unit=(u.hourangle, u.deg))
-    >>> c = SkyCoord('00:42.5 +41:12', frame='icrs', unit=(u.hourangle, u.deg))
```
Comment (taldcroft) [inline docs/coordinates/index.rst]: Removed explicit `frame='icrs'` to illustrate the default.

File: docs/coordinates/index.rst [hunk 2]
```diff
@@ -47,16 +47,35 @@ object:
 
 - Coordinate values can be provided either as unnamed positional arguments or
   via keyword arguments like ``ra``, ``dec``, ``l``, or ``b`` (depending on the frame).
-- Coordinate frame value is optional and can be specified as a positional
```
Comment (taldcroft) [inline docs/coordinates/index.rst]: Remove reference to specifying `frame` as a positional arg.

**Comment by taldcroft** [comment]
Are we allowing documentation improvements in bug release updates?  If so this might be a candidate for 1.0.1 (which doesn't appear to be released yet).

**Comment by astrofrog** [comment]
@taldcroft - in the past we've ruled that yes, documentation improvements can go in bug fix releases :)

**Comment by astrofrog** [comment]
Very nice, this is already a lot more readable! :+1:

**Comment by taldcroft** [comment]
@eteq - are you good with this doc update for coordinates Getting Started?  Hoping to get this into 1.0.1.

**Comment by astrofrog** [comment]
@embray - can we include this in 1.0.1 too?

**Comment by embray** [comment]
I don't see why not, so long as it's merged in time.

**Comment by taldcroft** [comment]
@astrofrog and I think it is a definite improvement, so I'm merging.  If @eteq has further tweaks then these can always go in the next release, or 1.0.1 if he is fast enough.  :smile:

**Comment by eteq** [comment]
That all seems fine to me, so it's fine that it just got merged.  Ironically, I think I'm not the best person to do these sections because I know it *too* well...

Because it's relevant to @aastrofrog's comment: I'm trying to write up an astropy tutorial for coordinates showcasing a lot of this stuff.  I think that's probably the better format, anyway.

**Comment by embray** [comment]
At some point I'll have to read through it in full and see how well *I* understand it.  I haven't actually taken the time yet.

**Pull request merged**